### ISSUE TITLE
gn-parse makes designations into hybrids, tripping nomer to fail in resolution

### ISSUE THREAD

**Issue by jtmiller28** [globalbioticinteractions/nomer#107]
It appears that using gn-parse within nomer will assume that specificEpithets that begin with 'x' are hybrids resulting in a failure to match. 

Ex.    echo -e "\tCylindropuntia ×fosbergii\t(C.B.Wolf) Rebman, M.A.Baker & Pinkava" | nomer replace gn-parse | nomer append wfo

Outputs:   Cylindropuntia × fosbergii	(C.B.Wolf) Rebman, M.A.Baker & Pinkava	NONE		Cylindropuntia × fosbergii

If manually done without using gn-parse mapping is successful

Ex. echo -e "\tCylindropuntia ×fosbergii" | nomer append wfo

Outputs: Cylindropuntia ×fosbergii	HAS_ACCEPTED_NAME	WFO:0000632373	Cylindropuntia ×fosbergii	species		Cylindropuntia ×fosbergii	WFO:0000632373	species	http://www.worldfloraonline.org/taxon/wfo-0000632373

**Comment by jhpoelen** [comment]
@jtmiller28 thanks for sharing these details. 

Using your example, I found that gn-parse appears to insert a whitespace after the ```×```` character. 


```
$ echo -e "\tCylindropuntia ×fosbergii\t(C.B.Wolf) Rebman, M.A.Baker & Pinkava" | nomer append gn-parse
[main] INFO org.globalbioticinteractions.nomer.match.TermMatcherRegistry - using matcher [gn-parse]
	Cylindropuntia ×fosbergii	(C.B.Wolf) Rebman, M.A.Baker & Pinkava	SAME_AS		Cylindropuntia × fosbergii					
```

before/after:

```
Cylindropuntia ×fosbergii
Cylindropuntia × fosbergii
```

**Comment by jhpoelen** [comment]
and, it appears that WFO reports the name as ```Cylindropuntia ×fosbergii```. Currently, Nomer does does exact matches on ```nomer append wfo``` in an effort to separate pre-processing (e.g., name normalization/parsing) from matching. And, perhaps this example shows that application of taxonomic name formatting for hybrids appears different comparing  the "gn-parse"-universe and the WFO-universe. 

@dimus @qgroom I was wondering whether you have any insights in the application / rules of hybrids in botanical nomenclature.

**Comment by jhpoelen** [comment]
@jtmiller28 note that it appears that the gbif parser also adds the whitespace - 

```
$ echo -e "\tCylindropuntia ×fosbergii" | nomer append gbif-parse
[main] INFO org.globalbioticinteractions.nomer.match.TermMatcherRegistry - using matcher [gbif-parse]
	Cylindropuntia ×fosbergii	SAME_AS		Cylindropuntia × fosbergii				
```

**Comment by jhpoelen** [comment]
@jtmiller28 My main question is - why does the World of Flora Online use ```Cylindropuntia ×fosbergii``` whereas two pretty well known name parsers (globalnames, gbif parser) inject a whitespace in between the ```×``` and the name ```fosbergii```?

A possible (silly) workaround for facilitate matching against WFO using ```sed``` without adding built in name interpretation into Nomer's WFO can be:

```
$ echo -e "\tCylindropuntia × fosbergii" | sed 's/× /×/g' | nomer append wfo
	Cylindropuntia ×fosbergii	HAS_ACCEPTED_NAME	WFO:0000632373	Cylindropuntia ×fosbergii	species		Angiosperms | Caryophyllales | Cactaceae | Cylindropuntia | Cylindropuntia ×fosbergii	WFO:9949999999 | WFO:9000000088 | WFO:7000000098 | WFO:4000010332 | WFO:0000632373	phylum | order | family | genus | species	http://www.worldfloraonline.org/taxon/wfo-0000632373	
```

**Comment by jtmiller28** [comment]
@jhpoelen Yes I was talking to some plant systemist about this the other day, it does seem to be a proper/used way to name hybrids using genus x specificEpithet . Your insight on WFO having their specific way of attaching 'x' to the specificEpithet is definitely what causes the issue. 

I do see that workaround being effective considering this is more of a catalogue peculiarity and should probably be brought to their attention for a more complete mapping list, I'll make that consideration for when working with the data

**Comment by jhpoelen** [comment]
@jtmiller28 thanks for sharing your outcomes with experts on this. Do you think it would make WFO more usable if we "fix" the funny WFO provided names by inserted the whitespace on indexing / parsing?

**Comment by jhpoelen** [comment]
Also, there surely must be a reason why WFO chose to adopt this naming convention. Do you have any idea why?

**Comment by jhpoelen** [comment]
@jtmiller28 very much would like to hear the backstory on the WFO hybrid naming conventions. Thanks for being thorough on this. Formatting names are pretty important I think, especially for integrating lots of datasets.

**Comment by dimus** [comment]
Hm, I imagine the problem is with old Sala gnparser @jhpoelen:

http://parser.globalnames.org/?format=html&names=Cylindropuntia+%C3%97fosbergii&with_details=on

Go-written current parser does parse hybrids correctly
